pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/CredentialServiceImpl.scala

```diff
@@ -1032,6 +1043,20 @@ private class CredentialServiceImpl(
     } yield record
   }
 
+  //   what needs to happen:
+  //       get the last status list of this wallet that was created
+  //        check its size and last used index
+  //          if last used index is less then size, that is the status list we will use, status list index is last_used_index + 1
+  //          if the last used index is equals to the size create new status list for this wallet, and use that, status list_index is 0
+  //
+
+  private[this] def allocateNewCredentialInStatusListForWallet(walletId: WalletId, recordId: DidCommID) = {
+    for {
+      lastStatusList <- credentialStatusListRepository.getLatestOfTheWallet(walletId)
```

Comment:
Do we agree that the process of (1) retrieving `lastIndex` and (2) updating to `lastIndex + 1` should be done atomically from a DB perspective?